lib/evmone/analysis.cpp

```diff
@@ -91,22 +84,23 @@ code_analysis analyze(evmc_revision rev, const uint8_t* code, size_t code_size)
 
         case ANY_SMALL_PUSH:
         {
-            const auto push_size = size_t(opcode - OP_PUSH1 + 1);
+            const auto push_size = static_cast<size_t>(opcode - OP_PUSH1) + 1;
             const auto push_end = code_pos + push_size;
 
-            uint8_t value_bytes[8]{};
-            auto insert_pos = &value_bytes[sizeof(value_bytes) - push_size];
-
-            // TODO: Consier the same endianness-specific loop as in ANY_LARGE_PUSH case.
+            uint64_t value = 0;
+            auto insert_bit_pos = (push_size - 1) * 8;
             while (code_pos < push_end && code_pos < code_end)
-                *insert_pos++ = *code_pos++;
-            instr.arg.small_push_value = load64be(value_bytes);
+            {
+                value |= uint64_t{*code_pos++} << insert_bit_pos;
```

Comment:
Could it still be in a separate helper function maybe?

Reply:
The `load64be` was only doing `bswap`. I would need to extract whole `case:`.

Reply:
I pushed an example how it could look like, but it is up to 30% slower.

Reply:
Wow then let's leave it, if it's so much slower (why, can't be inlined sometimes?)

`push_end` change looks fine, though

Reply:
I think returning multiple values and using tuple and `std::tie` messes it up. But I'm only guessing. I'd need much more time to investigate.